packages/rum/src/viewCollection.ts

```diff
@@ -175,3 +201,15 @@ function trackTimings(lifeCycle: LifeCycle, callback: (timings: Timings) => void
   )
   return { stop: stopPerformanceTracking }
 }
+
+function trackLoadDuration(lifeCycle: LifeCycle, id: string, callback: (loadDurationValue: number) => void) {
```

Comment:
This function should use `waitIdlePageActivity` to compute the load duration. You should remove the `VIEW_LOAD_COMPLETED` event, and any 'view load' related code in the user action collection.

Reply:
Then how do you know if you need to cancel the current UA while starting to load a new view ? Having the two process totally separated might hide the functional link between UA and page view. I am not super fan of that

Reply:
The collectUserAction module should expose a `cancelCurrentUserAction` function (as stated [here](https://github.com/DataDog/browser-sdk/pull/388#pullrequestreview-405046055)). AFAIK, this is the only link between the two processes.